npm/lit/src/mount.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  injectStylesBeforeElement,
+  StyleOptions,
+  getContainerEl,
+  setupHooks,
+} from '@cypress/mount-utils'
+import { LitElement, render, RootPart, TemplateResult } from 'lit'
+
+const DEFAULT_COMP_NAME = 'unknown'
+
+type LitElementConstructor<T> = new (...args: any[]) => T;
+
+export interface MountOptions extends
+  Partial<StyleOptions> {
+  log?: boolean
+}
+
+let componentInstance: LitElement | undefined
+
+const cleanup = () => {
+  componentInstance?.remove()
+}
+
+// Extract the component name from the object passed to mount
+const getComponentDisplayName = <T extends LitElement>(Component: LitElementConstructor<T>): string => {
+  if (Component.name) {
+    const [_, match] = /Proxy\<(\w+)\>/.exec(Component.name) || []
+
+    return match || Component.name
+  }
+
+  return DEFAULT_COMP_NAME
+}
+
+/**
+ * Mounts a Lit element inside the Cypress browser
+ *
+ * @param {LitElementConstructor<T>} Element Lit element being mounted
+ * @param {MountReturn<T extends SvelteComponent>} options options to customize the element being mounted
+ * @returns Cypress.Chainable<MountReturn>
+ *
+ * @example
+ * import Counter from './Counter'
+ * import { html } from 'lit'
+ * import { mount } from 'cypress/svelte'
+ *
+ * it('should render', () => {
+ *   mount(Counter, html`<my-counter .count=${32}></my-counter>`)
+ *   cy.get('button').contains(42)
+ * })
+ */
+export function mount<T extends LitElement> (
+  Component: LitElementConstructor<T>,
+  Template: TemplateResult,
+  options: MountOptions = {},
+): Cypress.Chainable<JQuery<HTMLElement>> {
+  return cy.then(() => {
+    const target = getContainerEl()
+
+    injectStylesBeforeElement(options, document, target)
+
+    const componentInstance = render(Template, target)
+
+    // by waiting, we are delaying test execution for the next tick of event loop
+    // and letting hooks and component lifecycle methods to execute mount
+    return cy.wait(0, { log: false }).then(() => {
+      if (options.log !== false) {
+        const mountMessage = `<${getComponentDisplayName(Component)} ... />`
+
+        Cypress.log({
+          name: 'mount',
+          message: [mountMessage],
+        }).snapshot('mounted').end()
+      }
+    })
+    .wrap({ component: componentInstance }, { log: false })
+    .get('[data-cy-root]')
+    .children()
+    .first()
+    .shadow()
```

Comment:
Does the bind make sense here, or can cause issue.

In theory it's not deterministic if mounting 2 components in 1 test (could return either depending on order).

Making the returned already bound to the shadow DOM is a better UX though.

Reply:
To me, it feels reasonable to expect a user to only provide a single component to test. I don't know how cypress's root element is managed though. One way to make sure is to create like an empty div to use as a render target and return its first child to be tested.

That div that was created can then be cleaned up with a `.remove()` easily on cleanup, without affecting anything else that might be in the cypress root.

Also, probably would be better to return out the actual component element from the mount here and not dive into the shadow DOM. User may want to test against the element's attribute/property or component might've been authored to not use shadow DOM. I think it should be up to the user to intentionally dive into the shadow DOM.